cockatrice/src/deck/deck_loader.cpp

```diff
@@ -80,6 +82,55 @@ bool DeckLoader::loadFromFile(const QString &fileName, FileFormat fmt, bool user
     return result;
 }
 
+bool DeckLoader::loadFromFileAsync(const QString &fileName, FileFormat fmt, bool userRequest)
+{
+    QFutureWatcher<bool> *watcher = new QFutureWatcher<bool>(this);
+
+    connect(watcher, &QFutureWatcher<bool>::finished, this, [this, watcher, fileName, fmt, userRequest]() {
+        const bool result = watcher->result();
+        watcher->deleteLater();
+
+        if (result) {
+            lastFileName = fileName;
+            lastFileFormat = fmt;
+            if (userRequest) {
+                updateLastLoadedTimestamp(fileName, fmt);
+            }
+            emit deckLoaded();
+        }
+
+        emit loadFinished(result);
+    });
+
+    QFuture<bool> future = QtConcurrent::run([=, this]() {
+        QFile file(fileName);
+        if (!file.open(QIODevice::ReadOnly | QIODevice::Text)) {
+            return false;
+        }
+
+        bool result = false;
```

Comment:
Can change this so the switch has the return in-line and default just returns false? Not sure if that's better than this